vms/avm/create_asset_tx.go

```diff
@@ -111,7 +134,60 @@ func (t *CreateAssetTx) SyntacticVerify(
 		}
 	}
 
-	if err := t.BaseTx.SyntacticVerify(ctx, c, txFeeAssetID, txFee, txFee, numFxs); err != nil {
+	if err := t.BaseTx.SyntacticVerify(ctx, c, txFeeAssetID, txFee, txFee, numFxs, epoch); err != nil {
+		return err
+	}
+
+	for _, state := range t.States {
+		if err := state.Verify(c, numFxs); err != nil {
+			return err
+		}
+	}
+	if !isSortedAndUniqueInitialStates(t.States) {
+		return errInitialStatesNotSortedUnique
+	}
+	return nil
+}
+
+func (t *CreateAssetTx) syntacticVerifyApricot(
+	ctx *snow.Context,
+	c codec.Manager,
+	txFeeAssetID ids.ID,
+	_ uint64,
+	txFee uint64,
+	numFxs int,
+	epoch uint32,
+) error {
+	switch {
+	case t == nil:
+		return errNilTx
+	case len(t.Name) < minNameLen:
+		return errNameTooShort
+	case len(t.Name) > maxNameLenApricot:
+		return errNameTooLong
+	case len(t.Symbol) < minSymbolLen:
+		return errSymbolTooShort
+	case len(t.Symbol) > maxSymbolLenApricot:
+		return errSymbolTooLong
+	case len(t.States) == 0:
+		return errNoFxs
+	case t.Denomination > maxDenomination:
+		return errDenominationTooLarge
+	case strings.TrimSpace(t.Name) != t.Name:
+		return errUnexpectedWhitespace
+	}
+
+	if !utf8.ValidString(t.Name) {
+		return errInvalidNameApricot
+	}
+
+	for _, r := range t.Symbol {
+		if r > unicode.MaxASCII || r < 32 {
+			return errInvalidSymbolApricot
+		}
+	}
+
```

Comment:
Is this too permissive @akly ?

Reply:
Can we exclude the max ascii value as well? `DEL` seems confusing and unnecessary.

`if r >= unicode.MaxASCII || r < 32 {`